### ISSUE TITLE
Compatibility with vcrpy >= 2.0

### ISSUE THREAD

**Issue by sathieu** [vmware/pyvmomi#748]
Following https://github.com/vmware/pyvmomi/pull/736#issuecomment-444623752

**Comment by jeking3** [comment]
http.client defines its behavior clearly:
```
class http.client.HTTPConnection(host, port=None, [timeout, ]source_address=None, blocksize=8192)

An HTTPConnection instance represents one transaction with an HTTP server. It should be instantiated passing it a host and optional port number. If no port number is passed, the port is extracted from the host string if it has the form host:port, else the default HTTP port (80) is used....
```

**Comment by jeking3** [comment]
I think vcrpy changed something internally and pyvmomi code is expecting a particular string no longer being returned:
```
  File "/home/jking/pyvmomi/.eggs/vcrpy-2.0.1-py3.6.egg/vcr/request.py", line 61, in port
    port = {'https': 443, 'http': 80}[parse_uri.scheme]
```
I'm looking into this now.

**Comment by jeking3** [comment]
I submitted a PR against vcrpy to resolve this.  The PR on vcrpy you mentioned was making corrupt URIs when the method was "CONNECT", so I fixed that.  Also the pyvmomi cassettes will need to be updated for those two tests, along with a new vcrpy version (like will be `vcrpy~=2.0.2`)

**Comment by jeking3** [comment]
That PR busted vcrpy's tests, so I'm going to focus on pyvmomi instead.  Perhaps we're formatting something badly.

**Comment by jeking3** [comment]
It was actually vcrpy, I have a PR out that fixes it now:

https://github.com/kevin1024/vcrpy/pull/415

Once that is merged AND released, will update pyvmomi to use `vcrpy~=2.0.2`.
